src/chainstate/stacks/db/mod.rs

```diff
@@ -1072,6 +1054,61 @@ impl StacksChainState {
         )
     }
 
+    /// Begin a transaction against the Clarity VM for initiating the genesis block
+    ///  the genesis block is special cased because it must be evaluated _before_ the
+    ///  cost contract is loaded in the boot code.
+    pub fn genesis_block_begin<'a>(
+        &'a mut self,
+        burn_dbconn: &'a dyn BurnStateDB,
+        parent_consensus_hash: &ConsensusHash,
+        parent_block: &BlockHeaderHash,
+        new_consensus_hash: &ConsensusHash,
+        new_block: &BlockHeaderHash,
+    ) -> ClarityTx<'a> {
+        let conf = self.config();
+        let headers_db = self.headers_state_index.sqlite_conn();
+        let clarity_instance = &mut self.clarity_state;
+
+        // mix burn header hash and stacks block header hash together, since the stacks block hash
+        // it not guaranteed to be globally unique (but the burn header hash _is_).
+        let parent_index_block =
+            StacksChainState::get_parent_index_block(parent_consensus_hash, parent_block);
+
+        let new_index_block =
+            StacksBlockHeader::make_index_block_hash(new_consensus_hash, new_block);
+
+        test_debug!(
+            "Begin processing Stacks block off of {}/{}",
```

Comment:
Can you update the `test_debug!` macros to indicate that this is the instantiation of the genesis block, and not just any old block?

Reply:
Yep, addressed in 983796523